crates/matrix-sdk-crypto/src/session_manager/group_sessions/share_strategy.rs

```diff
@@ -125,75 +125,119 @@ pub(crate) async fn collect_session_recipients(
     // This is calculated in the following code and stored in this variable.
     let mut should_rotate = user_left || visibility_changed || algorithm_changed;
 
-    let own_identity = store.get_user_identity(store.user_id()).await?.and_then(|i| i.into_own());
+    let mut devices: BTreeMap<OwnedUserId, Vec<DeviceData>> = Default::default();
+    let mut withheld_devices: Vec<(DeviceData, WithheldCode)> = Default::default();
 
-    for user_id in users {
```

Comment:
This feels like a big refactor, combined with behavioural changes. Please could we have the refactor and the behavioral change as separate commits?